### PR TITLE
LSR-874 	Support sections in BaseListTableView

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift [hunk 1]
```diff
@@ -91,4 +93,23 @@ public class BaseListView: BaseScreenletView {
 		
 		return rows[key]!
 	}
+	
+	internal func areThereMoreRows(newRows: [String : [AnyObject]]) -> Bool {
```
Comment (jmnavarro) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: renama as hasMoreRows?

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift [hunk 2]
```diff
@@ -91,4 +93,23 @@ public class BaseListView: BaseScreenletView {
...
+		if newRows.count == 0 {
+			return false
+		}
+		
+		let newRowsCount = newRows.values.reduce(0) { $0 + $1.count }
+		let isFirstPage = rows.values.reduce(0) { $0 + $1.count } == 0
```
Comment (jmnavarro) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: why count everything to figure out whether or not it's first page?

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift [hunk 3]
```diff
@@ -19,7 +19,7 @@ public class BaseListView: BaseScreenletView {
 	
 	public static let DefaultSection = ""
 	
-	public var streamMode: Bool = false
+	internal var streamMode: Bool = false
```
Comment (jmnavarro) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: do you need there variables to store the same flag? what if they're inconsistent? can you do it with less variables? even with only one?

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift (hunk 2 above)
Comment (victorg1991) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: We can only know if we are in the first page verifying that there are no rows in any section

File: ios/Framework/Core/Base/BaseListScreenlet/TableView/BaseListTableView.swift [hunk 4]
```diff
@@ -51,20 +51,27 @@ public class BaseListTableView: BaseListView, UITableViewDataSource, UITableView
...
+		if  moreRowsThanExpected {
 			turnStreamModeOn()
-			onAddedRows(lastCount: oldRows.count)
-		} else if rows.count < oldRows.count {
-			deleteRows(from: rows.count, to: oldRows.count)
+			onAddedRows(oldRows)
+		}
+		else if lessRowsThanExpected {
+			deleteRows(from: rows[BaseListView.DefaultSection]!.count,
+						to: oldRows[BaseListView.DefaultSection]!.count, section: 0)
```
Comment (jmnavarro) [inline ios/Framework/Core/Base/BaseListScreenlet/TableView/BaseListTableView.swift]: section: 0???

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift (hunk 1 above)
Comment (victorg1991) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: 👍  in 2570d1e

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift (hunk 2 above)
Comment (victorg1991) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: 👍 in 2e9e2ba

File: ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift (hunk 3 above)
Comment (victorg1991) [inline ios/Framework/Core/Base/BaseListScreenlet/BaseListView.swift]: 👍  in f85e465

File: ios/Framework/Core/Base/BaseListScreenlet/TableView/BaseListTableView.swift (hunk 4 above)
Comment (victorg1991) [inline ios/Framework/Core/Base/BaseListScreenlet/TableView/BaseListTableView.swift]: 👍 in 452e03c

**Comment by jmnavarro** [comment]
👍  Great job!

**Pull request merged**